.github/workflows/ci-additional.yaml

```diff
@@ -116,68 +133,23 @@ jobs:
           python xarray/util/print_versions.py
       - name: Install mypy
         run: |
-          python -m pip install "mypy==1.13" --force-reinstall
+          python -m pip install mypy pytest-mypy-plugins
```

Comment:
We had pinned the mypy version to avoid `main` failing when a new mypy release came out. Can we restore this; if we want to propose removing the pin we can decouple that from this PR?

Reply:
No worries. I'll pin it to 1.15, which is the version that allowed me to remove some `# type: ignore` comments, as it contains a fix for the issue that previously required use of those comments.

Reply:
Yes I coincidentally upgraded it recently, so merging `main` should work